pkg/tfgen/generate_python.go

```diff
@@ -578,6 +573,24 @@ func (g *pythonGenerator) emitResourceType(mod *module, res *resourceType) (stri
 	w.Writefmtln("            opts)")
 	w.Writefmtln("")
 
+	w.Writefmtln("    @staticmethod")
+	w.Writefmt("    def get(resource_name, id, opts=None")
+	for _, prop := range res.outprops {
+		w.Writefmt(", %[1]s=None", pycodegen.PyName(prop.name))
+	}
+	w.Writefmtln("):")
+	g.emitGetDocstring(w, mod, res)
+	w.Writefmtln(
+		"        opts = pulumi.ResourceOptions(id=id) if opts is None else opts.merge(pulumi.ResourceOptions(id=id))")
```

Comment:
Would love to open an issue to track adding a version of `merge` that does this by default.  It's a shame that we'll have to encourage to write the kind of code themselves to use `merge` correctly.

Reply:
https://github.com/pulumi/pulumi/issues/3058